File: Reversi/Factories/BoardFactory.cs
using System;
using System.Drawing;
using Reversi.Board;

namespace Reversi.Factories
{
    public class BoardFactory
    {
        public Board.Board CreateBoard(Size size) => CreateBoard(size.Width, size.Height);


        public Board.Board CreateBoard(int x, int y)
        {
            Board.Board board = new Board.Board(x, y);

            CellFactory cellFactory = new CellFactory();

            int cellCount = x * y;
            for (int i = 0; i < cellCount; i++)
            {
                int currentX = i % board.Size.Width;
                int currentY = (int) Math.Floor((decimal) i / board.Size.Height);

                Cell cell = cellFactory.CreateCell(currentX, currentY);
                board.Cells[currentX, currentY] = cell;
            }

            return board;
        }
    }
}

File: Reversi-Website/Static/src/js/modules/reversi/board/board-factory.js
/**
 *
 * @param {Reversi.CellFactory} cellFactory
 * @constructor
 */
Reversi.BoardFactory = function (cellFactory) {

    /**
     *
     * @param {number} width: the width in cells
     * @param {number} height: the height in cells
     * @returns {Board}
     */
    this.createBoard = (width, height) => {
        let rows = [];
        for (let rowIndex = 0; rowIndex < height; rowIndex++) {
            let cells = [];
            for (let columnIndex = 0; columnIndex < width; columnIndex++) {
                let coords = new Reversi.Coords(rowIndex, columnIndex);

                cells[columnIndex] = cellFactory.createCell(coords);
            }
            rows[rowIndex] = cells;
        }

        let cells = [];
        rows.forEach(row => row.forEach(cell => cells[cell.coords.toString()] = cell));

        return new Reversi.Board(new Reversi.Size(width, height), cells);
    };
};